### DISCUSSION TITLE
Ability to improve user data model for complex structures

### DISCUSSION THREAD

**Discussion by drennvinn** [directus/directus#18617] [Feature Requests]
### Summary

I come from frameworks such as Symfony, .NET, etc., and I loved discovering Directus for my small and medium projects. However, I realized that there are some limitations in data modeling. Here, I would like to suggest user management.

It would be great to be able to extend the User entity in order to design different types of users beyond roles. This, for example, could allow for the creation of more specific registration forms and also improve task delegation.

### Basic Example

In order to illustrate the thought, here is an example diagram that could show a use case:
<img width="1451" alt="Capture d’écran 2023-05-16 à 11 21 15" src="https://github.com/directus/directus/assets/18407632/fc77c3f6-f495-45fd-bf65-e5124ea3cda7">

### Motivation

Instead of POST /users we could use POST /customer or /society , etc... which aims to create a new user in a more targeted manner.

### Detailed Design

I understand that implementing this feature would be quite complex, but it would significantly enhance user management for larger projects.
I am aware that this could affect the user directory module, but it could remain unchanged in its current form. The difference would be that the fields specific to each user type could be displayed on their respective editing pages.

### Requirements List

...

### Drawbacks

This functionality might pose challenges for small and medium projects if users prefer to stick with basic role-based management. 

### Alternatives

...

### Adoption Strategy

it could be implemented as an optional feature, allowing users to choose whether they want to utilize the extended user entity for more specific user management or continue with the basic role-based approach.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
